subprojects/plugins/src/main/java/org/gradle/api/internal/java/usagecontext/ConfigurationVariantMapping.java

```diff
@@ -59,7 +59,7 @@ public void addAction(Action<? super ConfigurationVariantDetails> action) {
 
     public void collectUsageContexts(final ImmutableCollection.Builder<UsageContext> outgoing) {
         if (!outgoingConfiguration.isTransitive()) {
-            DeprecationLogger.warnOfChangedBehaviour("Publication ignores 'transitive = false' at configuration level.", "Consider using 'transitive = false' at the dependency level if you need this to be published.")
+            DeprecationLogger.warnOfChangedBehaviour("Publication ignores 'transitive = false' at configuration level - this behavior is deprecated.", "Consider using 'transitive = false' at the dependency level if you need this to be published.")
```

Comment:
Maybe we can just always add "This behavior is deprecated." in the factory method for this nag